docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,513 @@
+Basic Usage
+===========
+
+In our opinion the best way to understand how some stuff works is by showing some examples so this section gather all knowledge form previous chapters.
+Assume that we are implementing managing countries in our system.
+Let the fun begin.
+
+Describing feature
+------------------
+First of all let's start with writing our feature file the most crucial thing here is ubiquitous language. We don't want to specify how something is done,
+it doesn't really matter in case of business.
+``features/addressing/managing_countries/adding_country.feature``
+.. code-block:: feature
+    @managing_countries
+    Feature: Adding a new country
+        In order to sell my goods to different countries
+        As an Administrator
+        I want to add a new country to the store
+
+        Background:
+            Given I am logged in as an administrator
+
+        @ui
+        Scenario: Adding country
+            Given I want to add a new country
+            When I choose "France"
+            And I add it
+            Then I should be notified that it has been successfully created
+            And the country "France" should appear in the store
+
+Pay special attention to form of those sentences, they are specially constructed that way to hide details like ``I click on select box and i click on France country, I see France country in box with css .table``.
+The side effect of this is that we got more generic steps and when we want to change way of testing this feature for example in domain context it will be extremely easy all we need to do is adding ``@domain`` tag and of course implement proper steps in domain context.
+
```

Comment:
Pay attention to the form of these sentences. From the developer point of view they are hiding the details of the feature's implementation. Instead of describing 'When I click on the select box And I choose France from the dropdown Then I should see the France country in the table" - we are using sentences that are less connected with the implementation, but more focused on the effects of our actions.
A side effect of such approach  is that it results in steps being really generic, therefore if we want to add another way of testing this feature for instance in the domain context, it will be extremely easy to apply. We just need to add a different tag (in this case "@domain") and of course implement the proper steps in the domain context of our system.
To be more descriptive let's imagine that we want to check if a country is added properly in two ways. First we are checking if the adding works via frontend, so we are implementing steps that are clicking, opening pages, filling fields on forms and similar, but also we want to check this action regardlessly of the frontend, for that we need the domain, which allows us to perform actions only in the database.